core/src/main/java/io/grpc/Metadata.java

```diff
@@ -362,17 +364,30 @@ private String toStringInternal() {
    *
    * <p>Only the following ASCII characters are allowed in the names of keys:
    * <ul>
-   *   <li>letters, i.e., {@code a-z} and {@code A-Z}</li>
-   *   <li>digits, i.e., {@code 0-9}</li>
-   *   <li>dash, i.e., {@code -}</li>
-   *   <li>underscore, i.e., {@code _}</li>
+   *   <li>letters: {@code a-z}</li>
```

Comment:
Nonsense. Any detail we can easily hide is a good idea to hide. If we are unable to (like HTTP doesn't support it) or it is overly burdensome or counter-productive, then sure. But exposing an internal detail to prevent adding two lines of code with no real performance issue seems very short-sighted.